configure.ac

```diff
@@ -2026,101 +2027,118 @@ if test "X$THREADSAFE" = "Xyes"; then
     ## and/or a library path.  If the library path is specified then it must
     ## be preceded by a comma.
     ##
-    ## Thread-safety in HDF5 only uses Pthreads via configure, so the
-    ## default is "check", though this only has an effect when
-    ## --enable-threadsafe is specified.
+    ## The default is to use Pthreads when building with the Autotools, unless
+    ## we're building w/ MinGW.
     AC_SUBST([HAVE_PTHREAD]) HAVE_PTHREAD=yes
     AC_ARG_WITH([pthread],
                 [AS_HELP_STRING([--with-pthread=DIR],
                                 [Specify alternative path to Pthreads library when
-                                thread-safe capability is built.])],,
-                [withval=check])
-
-    case "$withval" in
-      check | yes)
-        AC_CHECK_HEADERS([pthread.h],, [unset HAVE_PTHREAD])
-        if test "x$HAVE_PTHREAD" = "xyes"; then
-          AC_CHECK_LIB([pthread], [pthread_self],, [unset HAVE_PTHREAD])
+                                thread-safe capability is built. Set this to
+                                'yes' or the location of Pthreads when building
+                                with MinGW and you would rather use Pthreads
+                                than Win32 threads.])],,
+                [withval=default])
+
+    ## If we're on MinGW, we want to use Win32 threads unless the builder
+    ## explicitly specifies --with-pthreads=(yes | path(s))
+    mingw_use_win32_threads="no"
+    if test -n "$HDF_MINGW" -a "$HDF_MINGW" = "yes" ; then
+        # Default or no --> Win32 threads
+        if test "$withval" = "default" -o "$withval" = "no" ; then
+            mingw_use_win32_threads="yes"
+            unset HAVE_PTHREAD
+            AC_DEFINE([HAVE_WIN_THREADS], [1], [Define to 1 if you have win32 threads])
         fi
-        ;;
-      no)
-        AC_MSG_ERROR([Must use Pthreads with thread safety])
-        ;;
-      *)
+    fi
```

Comment:
The above tests are new, and check to see if we're on MinGW and if someone specified --enable-pthreads=(yes or paths).

Everything below is just tabbing the existing code over by one.